### ISSUE TITLE
Implement scrambling functions

### ISSUE THREAD

**Issue by scottbedard** [scottbedard/cube#1]
Cubes need to be scramble-able, here is the proposed API for this feature.

```js
const cube = new Cube(3);

// generate a scramble, and return that scramble as an array of turns this function 
// will not modify the state of the cube, or the currentScramble state
cube.generateScramble();

// generate and execute a scramble, and return that scramble as an array of turns
cube.scramble();

// get the cube's current scramble. this value will be an empty array before a
// scramble is executed
cube.getCurrentScramble();
```

#### Basic requirements

- Scrambles should not generate "canceling" turns. For example, the turns `F` followed by `F-` would not change the state of the cube, and therefor should not be part of a scramble. Allowing opposite-face turns (while unlikely) could cause the same result. For example, `F D F-` would be equivalent to a single `D` turn. One way to prevent this would be to have each turn followed by an intersecting turn. So something like `F R F-` would be fine, because there is no way to produce that state with less than 3 turns.

- A node script should be exposed to generate scrambles. This will make it easier for a backend to generate and store scrambles, and know that solves submitted from the frontend were from genuine scrambles. This will make it a bit harder for anyone to cheat at games built off of this.

**Comment by scottbedard** [comment]
Reopening, we need need a node export

**Comment by scottbedard** [comment]
Closed as of [0.2.2](https://github.com/scottbedard/cube/releases/tag/0.2.2)
